### PR TITLE
add search option to ignore segment

### PR DESCRIPTION
**Pull request by aoiasd**
related:https://github.com/milvus-io/milvus/issues/21836
Signed-off-by: aoiasd [zhicheng.yue@zilliz.com](mailto:zhicheng.yue@zilliz.com)

### CODE REVIEW

**Comment by aoiasd** [comment]
rerun ut

File: internal/querynode/shard_cluster.go [hunk 1]
```diff
@@ -1106,3 +1107,57 @@ func (sc *ShardCluster) GetSegmentInfos() []shardSegmentInfo {
 func (sc *ShardCluster) getVersion() int64 {
 	return sc.version
 }
+
+func getSearchWithStreamingFunc(searchCtx context.Context, req *querypb.SearchRequest, node *QueryNode, qs *queryShard, nodeID int64, streamingResults **internalpb.SearchResults) withStreaming {
```
Comment (congqixia) [inline internal/querynode/shard_cluster.go:1111]: Please do not use `**internalpb.SearchResults`. We could modify `WithStreaming` signature to fix this problem

**Review by congqixia — CHANGES_REQUESTED** [review]

**Comment by xiaofan-luan** [comment]
looks good!
we should bring this in for 2.2.4

**Comment by aoiasd** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm
/kind enhancement

**Comment by xiaofan-luan** [comment]
Good job!
Hopefully we can get this feature work in 2.2.4!

**Pull request merged**